resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/MountedDependencyManagerImpl.scala

```diff
@@ -0,0 +1,320 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes.submit.v2
+
+import java.io.{File, FileOutputStream, StringWriter}
+import java.util.Properties
+import javax.ws.rs.core.MediaType
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.google.common.base.Charsets
+import com.google.common.io.{BaseEncoding, Files}
+import io.fabric8.kubernetes.api.model.{ConfigMap, ConfigMapBuilder, Container, ContainerBuilder, EmptyDirVolumeSource, PodBuilder, Secret, SecretBuilder, VolumeMount, VolumeMountBuilder}
+import okhttp3.RequestBody
+import retrofit2.Call
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+import org.apache.spark.{SparkException, SSLOptions}
+import org.apache.spark.deploy.kubernetes.CompressionUtils
+import org.apache.spark.deploy.kubernetes.config._
+import org.apache.spark.deploy.kubernetes.constants._
+import org.apache.spark.deploy.rest.kubernetes.v1.{KubernetesCredentials, KubernetesFileUtils}
+import org.apache.spark.deploy.rest.kubernetes.v2.{ResourceStagingServiceRetrofit, RetrofitClientFactory, StagedResourceIdentifier}
+import org.apache.spark.util.Utils
+
+private[spark] trait MountedDependencyManager {
+
+  /**
+   * Upload submitter-local jars to the resource staging server.
+   * @return The resource ID and secret to use to retrieve these jars.
+   */
+  def uploadJars(): StagedResourceIdentifier
+
+  /**
+   * Upload submitter-local files to the resource staging server.
+   * @return The resource ID and secret to use to retrieve these files.
+   */
+  def uploadFiles(): StagedResourceIdentifier
+
+  def configurePodToMountLocalDependencies(
+    driverContainerName: String,
+    initContainerSecret: Secret,
+    initContainerConfigMap: ConfigMap,
+    originalPodSpec: PodBuilder): PodBuilder
+
+  def buildInitContainerSecret(jarsSecret: String, filesSecret: String): Secret
+
+  def buildInitContainerConfigMap(
+    jarsResourceId: String, filesResourceId: String): ConfigMap
+
+  /**
+   * Convert the Spark jar paths from their locations on the submitter's disk to
+   * the locations they will be downloaded to on the driver's disk.
+   */
+  def resolveSparkJars(): Seq[String]
+
+  /**
+   * Convert the Spark file paths from their locations on the submitter's disk to
+   * the locations they will be downloaded to on the driver's disk.
+   */
+  def resolveSparkFiles(): Seq[String]
+}
+
+private[spark] class MountedDependencyManagerImpl(
```

Comment:
javadoc on this -- it's an implementation of `MountedDependencyManager` that is backed by a `ResourceStagingService`